src/main/java/org/apache/xml/security/utils/resolver/ResourceResolverContext.java

```diff
@@ -51,4 +57,12 @@ public ResourceResolverContext(Attr attr, String baseUri, boolean secureValidati
         return properties;
     }
 
+    public boolean isSafeURIToResolve() {
+        if (allowUnsafeResourceResolving) {
+            return true;
+        }
+        boolean forbiddenURI = (uriToResolve != null && (uriToResolve.startsWith("file:") || uriToResolve.startsWith("http:")))
+                || (baseUri != null && (baseUri.startsWith("file:") || baseUri.startsWith("http:")));
```

Comment:
For the baseUri, you can make this check more precise by only checking relative URIs. This way you won't trigger false cases where a baseURI has been set for resolving some other URI but not an unsafe reference URI. Something like:

`if (baseUri != null && !baseUri.isEmpty() && baseUri.charAt(0) != '#') {
  // now check if it is file or http ...`